plugins/modules/aci_interface_config.py

```diff
@@ -0,0 +1,440 @@
+        description=dict(type="str", aliases=["descr"]),
+        node=dict(type="int", aliases=["node_id"]),
+        pc_member=dict(type="str", aliases=["port_channel_member"]),
+        port_type=dict(type="str", default="access", choices=["access", "fabric"]),
+        role=dict(type="str", choices=["leaf", "spine"], aliases=["node_type"]),
+        admin_state=dict(type="str", choices=["up", "down"]),
+        interface_type=dict(
+            type="str",
+            default="switch_port",
+            choices=["switch_port", "pc_or_vpc", "fc", "fc_port_channel", "leaf_fabric", "spine_access", "spine_fabric"],
+        ),
+        interface=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["node", "interface"]],
+            ["state", "present", ["node", "interface"]],
+        ],
+        mutually_exclusive=[("policy_group", "breakout")],
+    )
+
+    policy_group = module.params.get("policy_group")
+    breakout = module.params.get("breakout")
+    description = module.params.get("description")
+    node = module.params.get("node")
+    pc_member = module.params.get("pc_member")
+    port_type = module.params.get("port_type")
+    role = module.params.get("role")
+    admin_state = module.params.get("admin_state")
+    interface = module.params.get("interface")
+    interface_type = module.params.get("interface_type")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    try:
+        if node is not None and int(node) not in range(101, 4001):
+            aci.fail_json(msg="Node ID: {0} is invalid; it must be in the range of 101 to 4000.".format(node))
+
+        card, port_id, sub_port = (None, None, None)
+        if interface is not None:
+            interface_parts = interface.split("/")
+            if len(interface_parts) == 3:
+                card, port_id, sub_port = interface_parts
+            elif len(interface_parts) == 2:
+                card, port_id = interface_parts
+                sub_port = 0
+            else:
+                aci.fail_json(msg="Interface: {0} is invalid; The format must be either card/port/sub_port(1/1/1) or card/port(1/1)".format(interface))
+
+            if int(card) not in range(1, 65):
+                aci.fail_json(msg="Card ID: {0} is invalid; it must be in the range of 1 to 64.".format(card))
+
+            if int(port_id) not in range(1, 129):
+                aci.fail_json(msg="Port ID: {0} is invalid; it must be in the range of 1 to 128.".format(port_id))
+
+            # Sub Port ID - 0 is default value
+            if int(sub_port) not in range(0, 17):
+                aci.fail_json(msg="Sub Port ID: {0} is invalid; it must be in the range of 0 to 16.".format(sub_port))
+    except (TypeError, ValueError) as error:
```

Comment:
Do we hit this scenario in the test cases?

Reply:
Do we hit the exception for TypeError as well?

Reply:
TypeError - may not occur! let me remove it.